src/OpenTelemetry/Metrics/MeterProviderBuilderExtensions.cs

```diff
@@ -107,9 +108,26 @@ public static MeterProviderBuilder AddView(this MeterProviderBuilder meterProvid
             if (metricStreamConfiguration is ExplicitBucketHistogramConfiguration histogramConfiguration)
             {
                 // Validate histogram boundaries
-                if (histogramConfiguration.Boundaries != null && !IsSortedAndDistinct(histogramConfiguration.Boundaries))
+                if (histogramConfiguration.Boundaries != null)
                 {
-                    throw new ArgumentException($"Histogram boundaries must be in ascending order with distinct values", nameof(histogramConfiguration.Boundaries));
+                    if (!HasValidHistogramBoundaries(histogramConfiguration.Boundaries))
+                    {
+                        throw new ArgumentException("Histogram bounds must be in ascending order with distinct values. double.NaN is not allowed.");
+                    }
+
+                    // Remove any infinity values from the histogram boundaries
+                    histogramConfiguration.Boundaries = histogramConfiguration.Boundaries.Where(x => !double.IsInfinity(x)).ToArray();
+
+                    // If empty bounds, use (-inf, +inf) as the single bucket
+                    if (!Enumerable.Any(histogramConfiguration.Boundaries))
+                    {
+                        histogramConfiguration.Boundaries = new double[] { double.NegativeInfinity, double.PositiveInfinity };
```

Comment:
Make the histogram of type HistogramSumCount?

Reply:
We should use `(-inf, +inf)` as the bucket only when the user explicitly provides the `Boundaries` as `new double[] { double.NegativeInfinity, double.PositiveInfinity }` in the view configuration.